pkg/controller/container-runtime-config/helpers.go

```diff
@@ -58,17 +61,21 @@ type tomlConfigCRIO struct {
 
 type updateConfigFunc func(data []byte, internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error)
 
-func createNewCtrRuntimeConfigIgnition(storageTOMLConfig, crioTOMLConfig []byte) igntypes.Config {
+// createNewIgnition returns an ignition config containing the new data configure by the CRs
+func createNewIgnition(configs map[string][]byte) igntypes.Config {
 	tempIgnConfig := ctrlcommon.NewIgnConfig()
 	mode := 0644
-	// Create storage.conf ignition
-	if storageTOMLConfig != nil {
-		storagedu := dataurl.New(storageTOMLConfig, "text/plain")
+	// Create ignitions
+	for filePath, data := range configs {
+		if data == nil {
```

Comment:
(Original question: Why is `data` treated specially here? It seems to me that it should in principle be possible to use this function to create an empty file, and `nil` is (sadly) a mostly valid empty slice in Go.

If the caller does not want a file created, the caller can just not include the key.)

---

Ah, this is actually used to avoid updating some of the files with variables like `registriesTOML []byte` initialized to `nil` by default.

That works, but it’s pretty non-obvious; if we are introducing a new helper, I’d prefer making this explicit in the callers. Let the callers build the `configs` map, using either a literal that contains the always-used files, or conditionally adding individual files to the map if they should be set up.

Non-blocking: If you want to keep the current interface, and explicitly document the “data == `nil` means the file is not included” semantics, that would be fine as well.

Reply:
Documented why this is being done. Fixed

Reply:
ACK. Could you move the `nil` documentation to the function doc string? It’s a property of the interface, not an internal implementation detail.